numpy/lib/npyio.py

```diff
@@ -721,110 +723,329 @@ def _savez(file, args, kwds, compress, allow_pickle=True, pickle_kwargs=None):
+                # error message matches this format.
+                raise TypeError(
+                    "usecols must be an int or a sequence of ints but "
+                    "it contains at least one element of type %s" %
+                    type(col_idx),
+                    ) from None
+        # Fall back to existing code
+        usecols = np.array([operator.index(i) for i in usecols_as_list],
+                           dtype=np.int32)
+
+    if ndmin not in [None, 0, 1, 2]:
+        raise ValueError(f'ndmin must be None, 0, 1, or 2; got {ndmin}')
+
+    if not isinstance(comment, str):
+        # assume comments are a sequence of strings
+        comments = tuple(comment)
+        comment = ''
+        # If there is only one comment, and that comment has one character,
+        # the normal parsing can deal with it just fine.
+        if len(comments) == 1:
+            if isinstance(comments[0], str) and len(comments[0]) == 1:
+                comment = comments[0]
+                comments = None
+    elif len(comment) > 1:
+        comments = (comment,)
+        comment = ''
     else:
-        types = []
-        packing = []
-        for field in dt.names:
-            tp, bytes = dt.fields[field]
-            flat_dt, flat_packer = _loadtxt_flatten_dtype_internal(tp)
-            types.extend(flat_dt)
-            flat_packing = flat_packer.args[0] if flat_packer else None
-            # Avoid extra nesting for subarrays
-            if tp.ndim > 0:
-                packing.extend(flat_packing)
-            else:
-                packing.append((len(flat_dt), flat_packing))
-        return (types, functools.partial(_loadtxt_pack_items, packing))
-
-
-def _loadtxt_pack_items(packing, items):
-    """Pack items into nested lists based on re-packing info."""
-    if packing is None:
-        return items[0]
-    elif packing is tuple:
-        return tuple(items)
-    elif packing is list:
-        return list(items)
+        comments = None
+
+    # comment is now either a 1 or 0 character string or a tuple:
+    if comments is not None:
+        assert comment == ''
+        # Note: An earlier version support two character comments (and could
+        #       have been extended to multiple characters, we assume this is
+        #       rare enough to not optimize for.
+        if quote != "":
+            raise ValueError(
+                "when multiple comments or a multi-character comment is "
+                "given, quotes are not supported.  In this case the quote "
+                "character must be set to the empty string: `quote=''`.")
     else:
-        start = 0
-        ret = []
-        for length, subpacking in packing:
-            ret.append(
-                _loadtxt_pack_items(subpacking, items[start:start+length]))
-            start += length
-        return tuple(ret)
+        # No preprocessing necessary
+        assert comments is None
```

Comment:
Here and a few lines up---do we really want a Python `assert` in the production code to only be used in lower-optimization builds? Or would it make sense to add simple errors/error messages for these code paths if genuinely invalid?